ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -527,13 +527,62 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+public func switchToLatest<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+	return SignalProducer<T, E> { sink, disposable in
+		let producerCompleted = Atomic(false)
+		let latestCompleted = Atomic(false)
+		
+		let completeIfNecessary: () -> () = {
+			if producerCompleted.value && latestCompleted.value {
+				sendCompleted(sink)
+			}
+		}
+
+		let latestDisposable = SerialDisposable()
+		disposable.addDisposable(latestDisposable)
+
+		producer.startWithSignal { signal, producerDisposable in
+			disposable.addDisposable(producerDisposable)
+			
+			signal.observe(
+				next: { signal in
+					latestDisposable.innerDisposable = nil
+					latestCompleted.value = false
+					
+					signal.startWithSignal { signal, signalDisposable in
+						latestDisposable.innerDisposable = signalDisposable
+						
+						materialize(signal).observe { event in
+							switch event {
+							case .Completed:
+								latestCompleted.value = true
+								completeIfNecessary()
+								
+							default:
+								sink.put(event)
+							}
+						}
+					}
+					
+					return
+				},
+				error: { error in
+					sendError(sink, error)
+				},
```

Comment:
I prefer to join these with the next line, a la the style of `-subscribeNext:error:completed:` used in RAC 2.x.